synapseclient/api/annotations.py

```diff
@@ -0,0 +1,39 @@
+"""
+The purpose of this module is to provide any functions that are needed to interact with
+annotations that are not cleanly provided by the synapseclient library.
+"""
+import json
+
+from dataclasses import asdict
+
+from typing import TYPE_CHECKING, Optional
+from synapseclient import Synapse
+
+if TYPE_CHECKING:
+    from synapseclient.models import Annotations
+
+
+def set_annotations(
+    annotations: "Annotations", synapse_client: Optional[Synapse] = None
+):
+    """Call to synapse and set the annotations for the given input.
+
+    :param annotations: The annotations to set. This is expected to have the id, etag, and annotations filled in.
+    :return: _description_
+    """
+    annotations_dict = asdict(annotations)
+
+    # TODO: Is there a more elegant way to handle this - This is essentially being used
```

Comment:
There is another dataclass library that adds some more functionality. Its [`asdict`](https://dataclass-wizard.readthedocs.io/en/latest/dataclass_wizard.html#dataclass_wizard.dumpers.asdict) function has an `exclude` argument that you can pass to leave out class attributes you don't want in your dictionary:

```
from dataclasses import dataclass
from dataclass_wizard.dumpers import asdict

@dataclass
class Foo:
    bar: str
    baz: int

foo = Foo('bar', 42)
print(asdict(foo, exclude=['baz']))

> {'bar': 'bar'}
```

You'd still be hard-coding the excluded values though. Not sure of any other ways aside from implementing an `asdict` method specific to the class that excludes attributes not used in the API.

Reply:
Good idea - but i'm starting to think in the other direction now. What I mean by this is specify only those things I want to include and have a dataclass -> json/dict mapping process to format the input for the REST API as the API is expecting.